core/build.sbt

```diff
@@ -176,6 +176,11 @@ setVersionFromTag := {
     .getOrElse("0.0.0-SNAPSHOT")}"""")
 }
 
+lazy val writeVersionToFile = taskKey[Unit]("Write project version to file")
```

Comment:
This helps to have a consistent version that will be used when running the tests for all the samples. The file will later be read by the build shell scripts.

The goal here is to avoid tasks that change files to break dynver (used as a fallback by the samples) since it generates another version if the git repository is dirty.

Reply:
Please remove this customization.

> The goal here is to avoid tasks that change files to break dynver

That's the reason why we should keep it as it was, the tasks should not change files to while running, otherwise, we want the CI to fail.

Reply:
It will fail anyway, but instead of failing with unresolved artifacts for the next sample, it would fail for the right reason if that is the case.

This is also relevant when running the Maven examples where we need the `CLOUDFLOW_VERSION` env since we don't have dynver there. Or is there an option for Maven samples?

Reply:
for Maven something like:
```bash
export CLOUDFLOW_VERSION=$(sbt -no-colors --error 'print version')
```
should make it work.

> It will fail anyway

If the git repo is not dirty the version is always the same.
But maybe I'm misunderstanding, can you elaborate? or maybe is faster to go through in a sync meeting?